src/br/com/zup/nossacasacodigo/cart/Cart.java

```diff
@@ -0,0 +1,46 @@
+package br.com.zup.nossacasacodigo.cart;
+
+import java.math.BigDecimal;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+import br.com.zup.nossacasacodigo.book.Book;
+
+public class Cart {
+	private Set<CartItem> items = new HashSet<>();
+	private static BigDecimal finalValue = BigDecimal.ZERO;
+
+	public void addToCart(Book book, int quantity) {
```

Comment:
O que acontece aqui se alguém passar uma quantidade negativa? Ou igual a zero?, ou então, se o livro for nulo?

Reply:
Com relação ao livro nulo seria possível considerando que para adicionar ao carrinho já deveriamos te-lo ? Com relação as quantidades, realmente faz sentido incrementar essa validação